api/plugins/check_instance_owner_plugin.go

```diff
@@ -0,0 +1,69 @@
+package plugins
+
+import (
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+	"net/http"
+	"strings"
+)
+
+const CheckInstanceOwnerPluginName = "CheckInstanceOwnerPlugin"
+
+type checkInstanceOwnerPlugin struct {
+	repository       storage.Repository
+	tenantIdentifier string
+}
+
+// NewCheckInstanceOwnerPlugin creates new plugin that checks the owner of the instance on bind requests
+func NewCheckInstanceOwnerPlugin(repository storage.Repository, tenantIdentifier string) *checkInstanceOwnerPlugin {
+	return &checkInstanceOwnerPlugin{
+		repository:       repository,
+		tenantIdentifier: tenantIdentifier,
+	}
+}
+
+// Name returns the name of the plugin
+func (p *checkInstanceOwnerPlugin) Name() string {
+	return CheckInstanceOwnerPluginName
+}
+
+// Bind intercepts bind requests and check if the instance owner is the same as the one requesting the bind operation
+func (p *checkInstanceOwnerPlugin) Bind(req *web.Request, next web.Handler) (*web.Response, error) {
+	ctx := req.Context()
+	callerTenantID := gjson.GetBytes(req.Body, "context."+p.tenantIdentifier).String()
+	pathSegments := strings.Split(req.URL.Path, "/")
+	instanceID := pathSegments[len(pathSegments)-3] // /v2/service_instances/:instance_id/service_bindings/:binding_id
+	byID := query.ByField(query.EqualsOperator, "id", instanceID)
+	object, err := p.repository.Get(ctx, types.ServiceInstanceType, byID)
+	if err != nil {
+		if err != util.ErrNotFoundInStorage {
+			return nil, util.HandleStorageError(err, string(types.ServiceInstanceType))
+		} else {
+			return next.Handle(req)
+		}
+	}
+	instance := object.(*types.ServiceInstance)
+
+	var instanceOwnerTenantID string
+	if instance.Labels != nil {
```

Comment:
Not sure we need this check as we always ensure that an instance is labelled, but I do understand that we want to be on the safe side.

What do you think about checking if the `instance.Labels` are nil, and if they are, then returning some error. This way we could catch if bugs which might have caused us not to label the instance. The error could say something like: _"Could not determine service instance owner"_

This would be returned even if there are labels, but the `tenantIdentifier` key is missing.